pkg/pushsync/pushsync.go

```diff
@@ -26,7 +26,7 @@ const (
 )
 
 type PushSyncer interface {
-	ChunkPusher(ctx context.Context, peer swarm.Address, ch swarm.Chunk) (*Receipt, error)
+	PushChunkToClosest(ctx context.Context, ch swarm.Chunk) (*Receipt, error)
```

Comment:
@jmozah golang convention on naming interfaces when it is a single-function interface, is that the name of the interface is the name of the function + `er` (or something in that manner, in a way that makes sense).
e.g.: 
- `fmt.Stringer` - is the interface for `String()` method
- `io.Closer` - interface for `Close() error` method
- `io.Reader` - interface for `Read()` method

The list goes on...
Can you please follow the same convention and rename the function to be consistent with the naming convention?
Also, I previously asked to rename the interface to `ChunkPusher`, I'm not sure why we ended up with `PushSyncer`

And another comment about the method name, is that you're leaking details about the internals of what the implementation is doing by the name. Personally I prefer shorter names that don't bind to some specific set of actions. We should treat implementation details as black-box on the package boundary

Reply:
In this case it is a bit more difficult to name this method, as we will have two methods in this interface, one for pushing to the closest peer (forwarding), and one to push directly to a specific peer (no forwarding). So, PushChunkToClosest does reveal the implementation detail, but it is also a requirement for that method.